### DISCUSSION TITLE
Customized BBN-Launcher and Node

### DISCUSSION THREAD

**Discussion by MeltedPixel** [bareboat-necessities/lysmarine_gen#315] [Show and tell]
I really liked the Idea of the BBN launcher.

It's current implementation is fine, but I felt like it could be cleaned up.

I re-wrote the NodeJS from scratch to instead load the dashboard directly as a standalone HTML file for easy editing and design. Additionally this also opens up the ability to quickly change and edit JS and CSS and even SASS if you feel so inclined.

I also wrote a Javascript Socket that talks back to NodeJS that will allow you to run applications from an OnClick event in the HTML itself.

![IMG_2521](https://github.com/bareboat-necessities/lysmarine_gen/assets/1194692/2a8ef646-4631-48a7-9bee-f8e82cb06a01)

**Comment by mgrouch** [comment]
Continue! You seem can make it more advanced. Do not forget to share code!

thanks

**Comment by MeltedPixel** [comment]
> Continue! You seem can make it more advanced. Do not forget to share code!
> 
> thanks

Of course, I will create a PR once I get everything cleaned up and commented.

**Comment by MeltedPixel** [comment]
Okay, my launcher is finished. I just need to write a script that generates the HTML for those that don't know how to code it or don't want to spend the time.

**Comment by Code-Out** [comment]
@MeltedPixel This looks great, I'm interested in doing something similar. Could you share your code on how you did this?

**Comment by Pearsonhenry2** [comment]
Following as I too am interested in editing the launcher. I have a 10.1" 1920x1080 screen so want something that uses all of that real estate better.

**Comment by MeltedPixel** [comment]
@Code-Out @Pearsonhenry2 Hey sorry for the late reply.

It actually wasn't too hard. I took the hardlink director for the current menu in node JS and pointed it to a custom HTML file. At that point I needed something to talk back to the OS when things were clicked on. So I just wrote a Javascript socket to handle passing that information.

I haven't done anything further with it, I will try and clean up the project and I will post the edits and scripts in here.

**Reply by Code-Out** [reply]
@MeltedPixel Thanks for coming back to the thread, I only really use HTML, CSS, PHP and a bit of JS so I'm not too clued up on node JS other than installing libraries so it would be good to check your code, I'm sure It'll click for me then.

Cheers,

**Comment by PtJudeRI** [comment]
That screen shot looks great!  I was wondering about refining the icons as well, as I am not using everything listed there.
